libs/test-setup/src/lib.rs

```diff
@@ -26,25 +26,75 @@ type AnyError = Box<dyn std::error::Error + Send + Sync>;
 
 const SCHEMA_NAME: &str = "prisma-tests";
 
-pub struct TestAPIArgs {
+pub struct TestApiArgs {
+    pub connector_tags: BitFlags<Tags>,
     pub test_function_name: &'static str,
-    pub test_tag: BitFlags<Tags>,
     pub test_features: BitFlags<Features>,
+    pub url_fn: &'static (dyn Fn(&str) -> String + Send + Sync),
+    pub provider: &'static str,
+}
+
+macro_rules! test_api_constructors {
+    ($(($connector:ident, $provider:literal),)*) => {
+        impl TestApiArgs {
+            $(
+                pub fn $connector(name: &'static str, tags: u16, features: u8) -> Self {
+                    let tags: BitFlags<Tags> = BitFlags::from_bits(tags).unwrap();
+                    let features: BitFlags<Features> = BitFlags::from_bits(features).unwrap();
+
+                    TestApiArgs {
+                        connector_tags: tags,
+                        test_function_name: name,
+                        test_features: features,
+                        url_fn: &urls::$connector,
+                        provider: $provider,
+                    }
+                }
+            )*
+        }
+    }
 }
 
-impl TestAPIArgs {
-    pub fn new(name: &'static str, tags: u16, features: u8) -> Self {
-        let tags: BitFlags<Tags> = BitFlags::from_bits(tags).unwrap();
-        let features: BitFlags<Features> = BitFlags::from_bits(features).unwrap();
-
-        TestAPIArgs {
-            test_function_name: name,
-            test_tag: tags,
-            test_features: features,
-        }
+test_api_constructors!(
+    (mssql_2017, "sql_server"),
+    (mssql_2019, "sql_server"),
+    (mysql_5_6, "mysql"),
+    (mysql, "mysql"),
+    (mysql_8, "mysql"),
+    (mysql_mariadb, "mysql"),
+    (postgres9, "postgresql"),
+    (postgres, "postgresql"),
+    (postgres11, "postgresql"),
+    (postgres12, "postgresql"),
+    (postgres13, "postgresql"),
+    (sqlite, "sqlite"),
+);
+
+impl TestApiArgs {
+    pub fn datasource_block(&self, url: &str) -> String {
+        format!(
+            r#"datasource db {{\nprovider = "{provider}"\nurl = "{url}" \n}}"#,
+            provider = self.provider,
+            url = url
+        )
     }
 }
 
+mod urls {
+    pub use super::mariadb_url as mysql_mariadb;
+    pub use super::mssql_2017_url as mssql_2017;
+    pub use super::mssql_2019_url as mssql_2019;
+    pub use super::mysql_5_6_url as mysql_5_6;
+    pub use super::mysql_8_url as mysql_8;
+    pub use super::mysql_url as mysql;
+    pub use super::postgres_10_url as postgres;
+    pub use super::postgres_11_url as postgres11;
+    pub use super::postgres_12_url as postgres12;
+    pub use super::postgres_13_url as postgres13;
+    pub use super::postgres_9_url as postgres9;
+    pub use super::sqlite_test_url as sqlite;
+}
```

Comment:
This is a bandaid — we can actually put the URL functions in that module if we are sure we want to go that way.